clients/admin-ui/src/features/datastore-connections/ConnectionsContainer.tsx

```diff
@@ -25,18 +32,32 @@ const ConnectionsContainer: React.FC = () => {
     )
   );
 
+  const { data: connectionTypesData } = useGetAllConnectionTypesQuery(
+    {
+      search: "",
+    },
+    { skip: connectionOptions.length > 0 }
+  );
+
+  const {
+    data: datastoreConnectionsData,
+    isFetching,
+    isLoading,
+    isSuccess,
+  } = useGetAllDatastoreConnectionsQuery(cachedFilters);
+
+  const hasData = datastoreConnectionsData!?.items?.length > 0;
```

Comment:
I don't think I've seen this pattern before with `!?` :eyes: So I'm curious what it does.

Reply:
! is [non-null assertion operator]  - its just saying to type checker that you're sure that a given variable is not null or undefined. The optional chaining operator ?. permits reading the value of a property located deep within a chain of connected objects without having to expressly validate that each reference in the chain is valid.

Reply:
Ah, yeah, I knew what they did separately - wasn't aware you could use both in this way though. Pretty neat!